cmd/bm-server/internal/account/file_message.go

```diff
@@ -138,22 +142,42 @@ func (r *fileRepo) MoveToBox(addr hash.Hash, srcBox, dstBox int, msgID string) e
 	return r.fs.Rename(srcPath, dstPath)
 }
 
-// Send a message to specific box
-// @TODO: This is a bit difficult: this is actually a bridge between the processing engine and the account storage
-// it assumes that both are using files. We must thus find a way to transfer a message from the processing to account
-// without assumptions. This probably means reading the message in-memory or something, and we don't like that either.
-// So we have to come up with a better way....
-func (r *fileRepo) SendToBox(addr hash.Hash, box int, msgID string) error {
+// Move message into the account system. THis is basically a bridge between the processing section, and the accounts.
+// I'm not sure if this needs to be here
+func (r *fileRepo) CreateMessage(addr hash.Hash, msgID string) error {
 	srcPath, err := message.GetPath(message.SectionProcessing, msgID, "")
 	if err != nil {
 		return err
 	}
 
-	dstPath := r.getPath(addr, filepath.Join(getBoxAsString(box), msgID))
-	// // If we have the inbox, the message is prefixed with the current timestamp (UTC). This allows us
-	// // sort on time locally and we can just fetch from a specific time (ie: fetch all messages since 20 minutes ago)
-	// if box == "inbox" {
-	// 	dstPath = r.getPath(addr, filepath.Join(box, fmt.Sprintf("%d-%s", time.Now().Unix(), msgID)))
-	// }
+	// Finally, move the message to our message directory
+	dstPath := r.getPath(addr, filepath.Join(messageDir, msgID))
 	return r.fs.Rename(srcPath, dstPath)
 }
+
+// RemoveMessage Removes a message complete from the account
+func (r *fileRepo) RemoveMessage(addr hash.Hash, msgID string) error {
```

Comment:
When removing a message we should check all their boxes to remove the symlinks too

Reply:
Probably the easiest way is to enumerate all the boxes and call RemoveFromBox

Reply:
True. Will fix